bin/incompatibility-check.js

```diff
@@ -60,21 +66,37 @@ function checkBreakingChanges(files) {
             }
 
             if (newSchema["meta:extends"] && Array.isArray(newSchema["meta:extends"]) && isMetaExtendsBroken(newSchema["meta:extends"], newSchema["allOf"])) { //check meta:extends against allOf
-                console.log('Warning: Incompatible "meta:extends" vs "allOf" found!!! The schemas inside "meta:extends" did not match those in "allOf".')
+                //no warning for now to make logging clean
+                //console.log('Warning: Incompatible "meta:extends" vs "allOf" found!!! The schemas inside "meta:extends" did not match those in "allOf".')
             }
 
             var differences = diff(originalSchema, newSchema);
             var brokenProperty = {};
-            if (differences && isPropertyRemoved(differences, brokenProperty)) { //check removed properties
-                errLogs.push(workingFile+' breaking changes found!!! Property "' + brokenProperty.name + '" can not be removed.\n');
+            if (differences) {
+                if (isPropertyRemoved(differences, brokenProperty)) { //check removed properties
+                    errLogs.push(workingFile+' breaking changes found!!! Property "' + brokenProperty.name + '" can not be removed.\n');
+                }
+
+                if (isPropertyChanged(differences, brokenProperty)) { //check changed properties
+                    if (brokenProperty.isFieldTypeBroken) {
+                        errLogs.push(workingFile+' breaking changes found!!! Data type of property "' + brokenProperty.name + '" can not be changed.\n');
+                    }
+                    if (brokenProperty.isStringFormatBroken) {
+                        errLogs.push(workingFile+' breaking changes found!!! String format of property "' + brokenProperty.name + '" can not be changed.\n');
+                    }
+                    if (brokenProperty.isMinimumBroken) {
```

Comment:
Min and Max values are used to determine the appropriate meta:xdmType. As long as the change does not change the binary type (make the field go from long to integer for example), then it is not a breaking change.

Reply:
this check is to ensure the data range can only be expanded.